warehouse/integrations/redshift/redshift.go

```diff
@@ -1294,10 +1294,16 @@ func (rs *Redshift) Cleanup(ctx context.Context) {
 				logfield.Error, err.Error(),
 			)
 		}
-		_ = rs.DB.Close()
 	}
 }
 
+func (rs *Redshift) Close() {
+	if rs.DB == nil {
+		return
+	}
```

Comment:
minor:
```
if rs.DB != nil {
    _ = rs.DB.Close()
}
```

CC: @achettyiitr common interface might help reduce some duplicate code per warehouse. Just keeping a note of it.